web/src/test/java/org/cbioportal/genome_nexus/web/AnnotationIntegrationTest.java

```diff
@@ -136,4 +155,86 @@ public void testLongDeletion()
+
+
+        HashMap annotationSummary = (HashMap) response.get("annotation_summary");
+        // transcriptConsequenceSummary is the object most often used for MAF
+        // annotation. It is a single transcript consequence that is deemed to
+        // be most relevant because it is the most impactful conequence to a
+        // canonical transcipt. If that is not available there are some other
+        // rules to decide which one is (see code)
+
+
+        // TODO: none of these fields work for GET endpoint, only works for POST
+        // HashMap transcriptConsequenceSummary = (HashMap) annotationSummary.get("transcriptConsequenceSummary");
+        // assertEquals(transcriptConsequenceSummary.get("transcriptId"), "ENST00000288602");
+        // assertEquals(transcriptConsequenceSummary.get("hugoGeneSymbol"), "BRAF");
+        // assertEquals(transcriptConsequenceSummary.get("hgvspShort"), "p.V600E");
+    }
+
+    @Test
+    public void testGenomicLocationSNVPOST() {
+        String genomicLocation = "7,140453136,140453136,A,T";
+
+        GenomicLocation[] genomicLocations = {
+            new GenomicLocation() {{
+                setChromosome(genomicLocation.split(",")[0]);
+                setStart(Integer.parseInt(genomicLocation.split(",")[1]));
+                setEnd(Integer.parseInt(genomicLocation.split(",")[2]));
+                setReferenceAllele(genomicLocation.split(",")[3]);
+                setVariantAllele(genomicLocation.split(",")[4]);
+            }}
+        };
+
+        List<Map<String, Object>> response = this.fetchVariantAnnotationByGenomicLocationPOST(genomicLocations);
+        assertEquals(response.get(0).get("hgvsg"), "7:g.140453136A>T");
+        HashMap annotationSummary = (HashMap) response.get(0).get("annotation_summary");
+        HashMap transcriptConsequenceSummary = (HashMap) annotationSummary.get("transcriptConsequenceSummary");
+        assertEquals(transcriptConsequenceSummary.get("transcriptId"), "ENST00000288602");
+        assertEquals(transcriptConsequenceSummary.get("hugoGeneSymbol"), "BRAF");
+        assertEquals(transcriptConsequenceSummary.get("hgvspShort"), "p.V600E");
+    }
+
+    @Test
+    public void testGenomicLocationNucleotideContextGET() {
+        String genomicLocation = "7,140453136,140453136,A,T";
+
+        Map<String, Object> response = this.fetchVariantAnnotationByGenomicLocationGET(genomicLocation);
+        HashMap nucleotideContext = (HashMap) response.get("nucleotide_context");
+        HashMap annotation = (HashMap) nucleotideContext.get("annotation");
+        String seq = (String) annotation.get("seq");
+        assertEquals("CAC", seq);
+    }
+
+    @Test
+    public void testGenomicLocationNucleotideContextPOST() {
+        String genomicLocation = "7,140453136,140453136,A,T";
+        GenomicLocation[] genomicLocations = {
+            new GenomicLocation() {{
+                setChromosome(genomicLocation.split(",")[0]);
```

Comment:
maybe do `String[] parts = genomicLocation.split(",")` above to avoid repetitive split calls, not a huge deal though.

Reply:
yeah there should prolly be some util function to convert genomicLocationString -> GenomicLocation